smarts/env/gymnasium/driving_smarts_competition_env.py

```diff
@@ -0,0 +1,299 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import logging
+import os
+import pathlib
+import typing
+from functools import partial
+
+import gymnasium as gym
+import numpy as np
+
+from envision.client import Client as Envision
+from envision.client import EnvisionDataFormatterArgs
+from smarts import sstudio
+from smarts.core.agent_interface import AgentInterface
+from smarts.core.controllers import ActionSpaceType
+from smarts.env.gymnasium.hiway_env_v1 import HiWayEnvV1, SumoOptions
+from smarts.env.multi_scenario_env import resolve_agent_interface
+
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.WARNING)
+
+SUPPORTED_ACTION_TYPES = (ActionSpaceType.RelativeTargetPose,)
+
+
+def driving_smarts_competition_v0_env(
+    scenario: str,
+    img_meters: int = 64,
+    img_pixels: int = 256,
+    action_space="RelativeTargetPose",
+    headless: bool = True,
+    seed: int = 42,
+    visdom: bool = False,
+    sumo_headless: bool = True,
+    envision_record_data_replay_path: typing.Optional[str] = None,
+):
+    """An environment with a mission to be completed by a single or multiple ego agents.
+
+    Observation space for each agent:
+
+        A ``smarts.core.sensors.Observation`` is returned as observation.
+
+    Action space for each agent:
+    .. note::
+
+        A ``smarts.core.controllers.ActionSpaceType.RelativeTargetPose``, which is a
+        sequence of [x-coordinate, y-coordinate, heading].
+
+        Type:
+
+        .. code-block:: python
+
+            gym.spaces.Box(
+                    low=np.array([-28, -28, -π]),
+                    high=np.array([28, 28, π]),
+                    dtype=np.float32
+                    )
+
+        .. list-table:: Table
+            :widths: 25 25
+            :header-rows: 1
+
+            * - Action
+              - Value range
+            * - Ego's next x-coordinate on the map
+              - [-28,28]
+            * - Ego's next y-coordinate on the map
+              - [-28,28]
+            * - Ego's next heading with respect to the map's axes
+              - [-π,π]
+
+    Reward:
+
+        Reward is distance travelled (in meters) in each step, including the termination step.
+
+    Episode termination:
+
+    .. note::
+
+        Episode is terminated if any of the following occurs.
+
+            1. Steps per episode exceed 800.
+
+            2. Agent collides, drives off road, drives off route, or drives on wrong way.
+
+    Solved requirement:
+
+    .. note::
+
+        If agent successfully completes the mission then ``info["score"]`` will
+        equal 1, else it is 0. Considered solved when ``info["score"] == 1`` is
+        achieved over 500 consecutive episodes.
+
+    :param scenario: Scenario name or path to scenario folder.
```

Comment:
Consider using Google styled dcostrings for standardized style and easy readability. Documentation standard suggestion is available [here]( https://smarts.readthedocs.io/en/rtd/resources/contributing.html#documentation).

Reply:
I will be cleaning up the docstrings tomorrow.